test/asmvariant.jl

```diff
@@ -0,0 +1,33 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+using Base.Test
+
+if Sys.ARCH === :x86_64 || Sys.Arch === :i386
```

Comment:
it might end up as `:i586` on some linux distros, possibly even `:i386` on some so maybe check for all ?